crates/nostr/src/nips/nip47.rs

```diff
@@ -201,16 +386,58 @@ impl Response {
     where
         S: AsRef<str>,
     {
-        match serde_json::from_str(json.as_ref()) {
-            Ok(response) => Ok(response),
-            Err(_err) => {
-                let json = json.as_ref().replace('\\', "");
-                Ok(serde_json::from_str(&json)?)
-            }
+        let template: ResponseTemplate = serde_json::from_str(json.as_ref())?;
+
+        if let Some(result) = template.result {
+            let result = match template.result_type {
+                Method::PayInvoice => {
+                    let result: PayInvoiceResponseResult = serde_json::from_value(result)?;
+                    ResponseResult::PayInvoice(result)
+                }
+                Method::MakeInvoice => {
+                    let result: MakeInvoiceResponseResult = serde_json::from_value(result)?;
+                    ResponseResult::MakeInvoice(result)
+                }
+                Method::LookupInvoice => {
+                    let result: LookupInvoiceResponseResult = serde_json::from_value(result)?;
+                    ResponseResult::LookupInvoice(result)
+                }
+                Method::GetBalance => {
+                    let result: GetBalanceResponseResult = serde_json::from_value(result)?;
+                    ResponseResult::GetBalance(result)
+                }
+            };
+
+            Ok(Self {
+                result_type: template.result_type,
+                error: template.error,
+                result: Some(result),
+            })
+        } else {
+            Ok(Self {
+                result_type: template.result_type,
+                error: template.error,
+                result: None,
+            })
         }
     }
 }
 
+impl<'de> Deserialize<'de> for Response {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Ok(Self::from_json(
+            Value::deserialize(deserializer)
+                .unwrap()
+                .to_string()
+                .as_str(),
+        )
+        .unwrap())
```

Comment:
The same here, replace the `.unwrap()` with `.map_err(serde::de::Error::custom)`